core/src/main/java/tc/oc/pgm/stats/StatsModule.java

```diff
@@ -0,0 +1,90 @@
+package tc.oc.pgm.stats;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import net.kyori.adventure.text.Component;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLFluentParser;
+import tc.oc.pgm.util.xml.XMLUtils;
+import tc.oc.pgm.variables.Variable;
+import tc.oc.pgm.variables.VariablesMatchModule;
+
+public class StatsModule implements MapModule<StatsMatchModule> {
+  private final List<MapVariableStat> variables;
+
+  private StatsModule(List<MapVariableStat> variables) {
+    this.variables = variables;
+  }
+
+  @Override
+  public @Nullable StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    final List<StatsMatchModule.VariableStat> variableStats = new ArrayList<>();
+    for (MapVariableStat mapVar : variables) {
+      if (mapVar.variable instanceof Variable.TrackedHolders<MatchPlayer>
+          && mapVar.variable instanceof Variable.DefinesDefault<MatchPlayer>) {
```

Comment:
if you need both of these, you're just asking for a DummyVariable, which i don't think is a good restriction

Reply:
Will address this in the similar comment below